File: examples/javascript/chat_basic.mjs
// Minimal Tara /chat call. No deps — uses Node 18+ native fetch.
//
// Usage:
//   export TARA_API_KEY=tara_sk_...
//   node chat_basic.mjs "What is Tratok?"

const TARA_API_BASE = "https://tara.tratok.com/api/v1";

async function chat(message) {
  const apiKey = process.env.TARA_API_KEY;
  if (!apiKey) {
    console.error("Set TARA_API_KEY in your environment first.");
    process.exit(1);
  }

  const response = await fetch(`${TARA_API_BASE}/chat.php`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${apiKey}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
      max_tokens: 512,
      temperature: 0.7,
    }),
  });

  if (!response.ok) {
    const text = await response.text();
    throw new Error(`HTTP ${response.status}: ${text}`);
  }
  return response.json();
}

const userInput = process.argv.slice(2).join(" ") || "What is Tratok in one sentence?";
const result = await chat(userInput);

if (!result.ok) {
  console.error(`Error: ${result.error?.message ?? "unknown"}`);
  process.exit(1);
}

console.log(result.reply);
console.log();
console.log(`--- tokens: ${result.usage.total_tokens}`);

File: examples/python/chat_basic.py
"""
Minimal Tara /chat example.

Usage:
    export TARA_API_KEY=tara_sk_...
    pip install -r requirements.txt
    python chat_basic.py
"""

from __future__ import annotations

import os
import sys

import requests

TARA_API_BASE = "https://tara.tratok.com/api/v1"


def chat(message: str) -> dict:
    api_key = os.environ.get("TARA_API_KEY")
    if not api_key:
        raise SystemExit("Set TARA_API_KEY in your environment first.")

    response = requests.post(
        f"{TARA_API_BASE}/chat.php",
        headers={
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json",
        },
        json={
            "message": message,
            "max_tokens": 512,
            "temperature": 0.7,
        },
        timeout=60,
    )
    response.raise_for_status()
    return response.json()


def main() -> None:
    user_input = " ".join(sys.argv[1:]) or "What is Tratok in one sentence?"
    result = chat(user_input)

    if not result.get("ok"):
        print(f"Error: {result.get('error', {}).get('message', 'unknown')}")
        sys.exit(1)

    print(result["reply"])
    print()
    print(f"--- tokens: {result['usage']['total_tokens']}")


if __name__ == "__main__":
    main()
